src/client/main.c

```diff
@@ -2353,6 +2353,22 @@ static size_t CL_WeaponModel_m(char *buffer, size_t size)
                        cl.configstrings[cl.frame.ps.gunindex + CS_MODELS]);
 }
 
+//Mako
+static size_t CL_Texture_m(char *buffer, size_t size)
+{
+	trace_t trace;
+	vec3_t forward, end;
+	vec3_t mins = { -1.f, -1.f, -1.f };
+	vec3_t maxs = { 1.f, 1.f, 1.f };
+	int mask = CONTENTS_SOLID | CONTENTS_WINDOW | CONTENTS_WATER | CONTENTS_SLIME | CONTENTS_LAVA;
+
+	AngleVectors(m_PlayerAngles, forward, NULL, NULL);
+	VectorMA(m_PlayerPos, 8192, forward, end);
+	CM_BoxTrace(&trace, m_PlayerPos, end, mins, maxs, cl.bsp->nodes, mask);
+
+	return Q_scnprintf(buffer, size, "%s", trace.surface->name);
```

Comment:
Can surface be NULL if you don't hit anything? Should probably add a check.

Reply:
It can't:
https://github.com/TotallyMehis/q2pro/blob/a66d1b68ae9de47e49e6d1951b92e2ee0d6d0dea/src/common/cmodel.c#L633

Reply:
👍